blob/service_test.go

```diff
@@ -298,20 +299,40 @@ func TestBlobService_Get(t *testing.T) {
 		{
 			name: "not included",
 			doFn: func() (interface{}, error) {
-				appBlob, err := blobtest.GenerateV0Blobs([]int{10}, false)
-				require.NoError(t, err)
-				blob, err := convertBlobs(appBlob...)
-				require.NoError(t, err)
-
 				proof, err := service.GetProof(ctx, 1,
 					blobsWithDiffNamespaces[1].Namespace(),
 					blobsWithDiffNamespaces[1].Commitment,
 				)
 				require.NoError(t, err)
-				return service.Included(ctx, 1, blob[0].Namespace(), proof, blob[0].Commitment)
+
+				// tamper with the header getter to get a random data hash at height 12345
+				service.headerGetter = func(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+					if height == 12345 {
+						return &header.ExtendedHeader{
+							RawHeader: header.RawHeader{
+								DataHash: []byte{0x01},
+							},
+						}, nil
+					}
+					return service.headerGetter(ctx, height)
+				}
+				// this blob was included in height 1, but we will check if it's included in height 2
+				return service.Included(
+					ctx,
+					12345,
+					blobsWithDiffNamespaces[1].Namespace(),
+					proof,
+					blobsWithDiffNamespaces[1].Commitment,
+				)
 			},
 			expectedResult: func(res interface{}, err error) {
-				require.NoError(t, err)
+				// Question for reviwers: for the case of included, it's here made not to return an error
```

Comment:
cc @vgonkivs

Reply:
so we can return the error now?

Reply:
Yes. This case is impossible, since verification does not require any network requests now.

Reply:
[fed1637](https://github.com/celestiaorg/celestia-node/pull/3821/commits/fed16378e5b055bdbc3637bd4125d943f142a0a1)